core/transaction-pool/src/api.rs

```diff
@@ -86,3 +94,82 @@ impl<T, Block> txpool::ChainApi for FullChainApi<T, Block> where
 		})
 	}
 }
+
+/// The transaction pool logic for light client.
+pub struct LightChainApi<T, F, Block> {
+	client: Arc<T>,
+	fetcher: Arc<F>,
+	_phantom: PhantomData<Block>,
+}
+
+impl<T, F, Block> LightChainApi<T, F, Block> where
+	Block: BlockT,
+	T: HeaderBackend<Block>,
+	F: Fetcher<Block>,
+{
+	/// Create new transaction pool logic.
+	pub fn new(client: Arc<T>, fetcher: Arc<F>) -> Self {
+		LightChainApi {
+			client,
+			fetcher,
+			_phantom: Default::default(),
+		}
+	}
+}
+
+impl<T, F, Block> txpool::ChainApi for LightChainApi<T, F, Block> where
```

Comment:
Any chance we could have a test for this? Or at least run transaction pool tests using `LightChainApi` or sth.

Reply:
There are tests in [`pool_maintainer.rs`](https://github.com/paritytech/substrate/pull/3833/files#diff-9db06f15e12d6c0feeaabcd41489533bR356) - `should_remove_transactions_from_the_light_pool`, `should_schedule_transactions_revalidation_at_light_pool` and `should_revalidate_transactions_at_light_pool`. Are they enough or do you want to test something else?

Reply:
That should do!